### PR TITLE
Changed namespace for HA devices to org.thingpedia

### PR DESCRIPTION
**Pull request by greentfrapp**
Also updated icons to be more general

### CODE REVIEW

File: io.home-assistant/thermostat.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+            throw new Error ('Sorry! Your device doesn\'t support checking temperature.');
+        }
+    }
+    // note: subscribe_ must NOT be async, or an ImplementationError will occur at runtime
+    subscribe_temperature() {
+        if (['binary_sensor', 'sensor'].includes(this.domain) && this.device_class === 'temperature') {
+            if (this.domain === 'sensor') {
+                let value = parseFloat(this.state.state);
+                return this._subscribeState(() => {
+                    return {state: undefined, value: value};
```
Comment (gcampax) [inline io.home-assistant/thermostat.js]: This code is incorrect, and will always return the same temperature. You need to access `this.state` inside the callback to `_subscribeState`, not outside.
(here and elsewhere)

File: io.home-assistant/thermostat.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        }
+        else {
+            throw new Error ('Sorry! Your device doesn\'t support checking temperature.');
+        }
+    }
+    
+    async get_hvac_state() {
+        if (this.domain === 'climate') {
+            let mode = this.state.hvac_mode;
+            let state = this.state.hvac_action;
```
Comment (gcampax) [inline io.home-assistant/thermostat.js]: Are the enums compatible between nest and home assistant?

File: com.nest/camera.js [hunk 3]
```diff
@@ -59,6 +59,10 @@ const CameraDevice = class NestCameraDevice extends Tp.BaseDevice {
         return stream;
     }
 
+    get_state() {
+        throw new Error ('Sorry! Your device doesn\'t support querying the state.');
```
Comment (gcampax) [inline com.nest/camera.js]: Do you plan to implement this? After all, we can change the state, so we should be able to query it.

File: org.thingpedia.iot.connectivity/manifest.tt [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+#_[description="Interface for Connectivity Sensor."]
+#[license="BSD-3-Clause"]
+#[license_gplcompatible=true]
+#[subcategory="home"]
+{
+  monitorable query state(out state : Enum(connected,disconnected))
+  #_[canonical="connectivity sensor state"]
+  #_[confirmation="the state of $__device"]
+  #_[formatted=["It is $state."]];
+}
```
Comment (gcampax) [inline org.thingpedia.iot.connectivity/manifest.tt]: Have we figured out how this sensor is supposed to be used? "Connectivity sensor" is not a commonly used expression in English...

File: org.thingpedia.iot.garage-door/dataset.tt [hunk 5]
```diff
@@ -0,0 +1,78 @@
+dataset @org.thingpedia.iot.garage-door language "en" {
```
Comment (gcampax) [inline org.thingpedia.iot.garage-door/dataset.tt]: Didn't we discuss merging door with garage door?

File: org.thingpedia.iot.security-camera/manifest.tt [hunk 6]
```diff
@@ -1,22 +1,31 @@
...
+#_[thingpedia_description="Interface for Security Camera and Related Sensors."]
+#_[description="Interface for Security Camera and Related Sensors."]
+#[license="BSD-3-Clause"]
+#[license_gplcompatible=true]
+#[subcategory="home"]
+{
+  monitorable query state(out state : Enum(detecting,not_detecting))
+  #_[canonical="sensor state"]
+  #_[confirmation="the state of $__device"]
+  #_[formatted=["The sensor is ${state:enum} movement."]];
```
Comment (gcampax) [inline org.thingpedia.iot.security-camera/manifest.tt]: What is this? This function does not make sense. The security camera interface should only have security camera features, and motion sensor features should go into the motion sensor interface. `com.nest.security_camera` can implement all of `org.thingpedia.iot.security-camera`, `org.thingpedia.iot.motion`, `org.thingpedia.iot.sound` and `org.thingpedia.iot.occupancy`.

File: org.thingpedia.iot.security-camera/manifest.tt [hunk 7]
```diff
@@ -1,22 +1,31 @@
...
   #[poll_interval=0ms]
   #[doc="trigger on any new event detected by the camera, providing the start time of the event and a small gif"];
 
-  action set_power(in req power: Enum(on,off) #_[prompt="Do you want to turn your camera on or off?"])
-  #_[canonical="set power on nest security camera"]
-  #_[confirmation="turn $power your Nest security camera"]
-  #_[confirmation_remote="turn $power $__person's Nest security camera"]
-  #[doc="change the streaming status of the camera"];
+  action set_power(in req power: Enum(on,off) #_[prompt="Do you want to turn your device on or off?"])
```
Comment (gcampax) [inline org.thingpedia.iot.security-camera/manifest.tt]: Why did you change the question? Wasn't the previous question better?

File: org.thingpedia.iot.speaker/manifest.tt [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+#_[thingpedia_description="Interface for Speaker."]
+#_[description="Interface for Speaker."]
+#[license="CC-0"]
+#[license_gplcompatible=true]
+#[subcategory="home"]
+{
+  monitorable query state(out state : Enum(on,off))
+  #_[canonical="switch state"]
+  #_[confirmation="the state of $__device"]
+  #_[formatted=["It is ${state}"]];
```
Comment (gcampax) [inline org.thingpedia.iot.speaker/manifest.tt]: What's the definition of speaker state in Home Assistant? With Bluetooth, you don't know on/off, all you know is connected/disconnected.

File: org.thingpedia.iot.speaker/manifest.tt [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+  action set_power(in req power: Enum(on,off) #_[prompt="Do you want to turn it on or off?"] #_[canonical="power"])
+  #_[canonical="turn your speaker on or off"]
+  #_[confirmation="turn ${power} $__device"]
+  #[confirm=true];
+  #[doc="turn on or off the TV"];
```
Comment (gcampax) [inline org.thingpedia.iot.speaker/manifest.tt]: TV copy-paste errors here and elsewhere

File: org.thingpedia.iot.speaker/manifest.tt [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+
+  action raise_volume()
```
Comment (gcampax) [inline org.thingpedia.iot.speaker/manifest.tt]: You use `volume_up` in dataset.tt. Did you lint?

File: org.thingpedia.iot.speaker/manifest.tt [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+  #[doc="set the tv volume"];
+
+  action set_sink()
+  #_[canonical="set audio sink on speaker"]
+  #_[confirmation="set $__device to be the default audio output"]
+  #_[confirmation_remote="set $__person's $__device to be their default audio output"]
+  #[doc="set this speaker as the default audio output device"];
+
+  action play_music()
+  #_[canonical="play music on bluetooth speaker"]
```
Comment (gcampax) [inline org.thingpedia.iot.speaker/manifest.tt]: No more "bluetooth" in the generic speaker interface

File: org.thingpedia.iot.thermostat/dataset.tt [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+    program := now => @org.thingpedia.iot.thermostat.humidity() => notify
+    #_[utterances=["is it humid?",
+                   "check if it is humid",
+                   "what does my humidity sensor say?",
+                   "how humid is it?",
+                   "what is the level of humidity?",
+                   "how high is the humidity?"]];
+
+    stream := monitor @org.thingpedia.iot.thermostat.humidity()
+    #_[utterances=["when the humidity changes"]];
```
Comment (gcampax) [inline org.thingpedia.iot.thermostat/dataset.tt]: We already have a humidity sensor, so we don't need this.

File: org.thingpedia.iot.thermostat/dataset.tt [hunk 13]
```diff
@@ -0,0 +1,130 @@
...
+                   "check if the room temperature is warm or cold",
+                   "what is the indoor temperature?",
+                   "how high is the indoor temperature?"]];
+
+    query := @org.thingpedia.iot.thermostat.temperature()
+    #_[utterances=["the state of the room temperature",
+                   "the room temperature",
+                   "how high the room temperature is",
+                   "if the inside temperature is warm or cold",
+                   "whether the room temperature is warm or cold"]];
```
Comment (gcampax) [inline org.thingpedia.iot.thermostat/dataset.tt]: Similarly, we already have a temperature sensor.

File: org.thingpedia.iot.tv/dataset.tt [hunk 14]
```diff
@@ -0,0 +1,217 @@
...
+
+    action (p_power :Enum(on,off)) := @org.thingpedia.iot.tv.set_power(power=p_power)
+    #_[utterances=["turn ${p_power} my television",
+                   "switch ${p_power} the television",
+                   "${p_power} the tv",
+                   "turn ${p_power} my tv",
+                   "set the power on my tv to ${p_power}"]];
+
+    action := @org.thingpedia.iot.tv.set_power(power=enum(on))
+    #_[utterances=["on my television",
```
Comment (gcampax) [inline org.thingpedia.iot.tv/dataset.tt]: We already noticed that "on my television" is a bad template... on/off are not verbs

File: org.thingpedia.iot.tv/manifest.tt [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+#_[thingpedia_description="Interface for TV."]
+#_[description="Interface for TV."]
+#[license="BSD-3-Clause"]
+#[license_gplcompatible=true]
+#[subcategory="home"]
+{
+  monitorable query state(out state : Enum(on,off))
+  #_[canonical="switch state"]
+  #_[confirmation="the state of $__device"]
+  #_[formatted=["It is ${state}"]];
```
Comment (gcampax) [inline org.thingpedia.iot.tv/manifest.tt]: I know you're fond of just using `state`, but in an interface with a lot of functions, it's unclear what `state` is. Also, copy-paste errors.

File: io.home-assistant/thermostat.js (hunk 1 above)
Comment (greentfrapp) [reply io.home-assistant/thermostat.js]: Fixed in latest commit

File: org.thingpedia.iot.garage-door/dataset.tt (hunk 5 above)
Comment (greentfrapp) [reply org.thingpedia.iot.garage-door/dataset.tt]: Merged in latest commit

File: org.thingpedia.iot.speaker/manifest.tt (hunk 11 above)
Comment (greentfrapp) [reply org.thingpedia.iot.speaker/manifest.tt]: Fixed in latest commit

File: org.thingpedia.iot.speaker/manifest.tt (hunk 9 above)
Comment (greentfrapp) [reply org.thingpedia.iot.speaker/manifest.tt]: Fixed in latest commit

File: org.thingpedia.iot.speaker/manifest.tt (hunk 10 above)
Comment (greentfrapp) [reply org.thingpedia.iot.speaker/manifest.tt]: Linted and fixed in latest commit

File: io.home-assistant/thermostat.js (hunk 2 above)
Comment (greentfrapp) [reply io.home-assistant/thermostat.js]: HA has a few more enum values that I added in the manifest.tt. I don't have to change anything on the nest side for this right?

File: com.nest/camera.js (hunk 3 above)
Comment (greentfrapp) [reply com.nest/camera.js]: Yup sure, where can I find the nest camera api?

File: org.thingpedia.iot.connectivity/manifest.tt (hunk 4 above)
Comment (greentfrapp) [reply org.thingpedia.iot.connectivity/manifest.tt]: Hmmm not really. I also asked about some of the other weirder sensors but Paulus doesn't seem to be sure either. Should we remove this?

File: org.thingpedia.iot.tv/dataset.tt (hunk 14 above)
Comment (greentfrapp) [reply org.thingpedia.iot.tv/dataset.tt]: Well... technically wouldn't we want Almond to recognize when a user says "on the tv"?

File: org.thingpedia.iot.thermostat/dataset.tt (hunk 12 above)
Comment (greentfrapp) [reply org.thingpedia.iot.thermostat/dataset.tt]: Fixed in latest commit

File: org.thingpedia.iot.thermostat/dataset.tt (hunk 13 above)
Comment (greentfrapp) [reply org.thingpedia.iot.thermostat/dataset.tt]: Fixed in latest commit

File: org.thingpedia.iot.security-camera/manifest.tt (hunk 6 above)
Comment (greentfrapp) [reply org.thingpedia.iot.security-camera/manifest.tt]: Dropped org.thingpedia.iot.security-camera for now

File: org.thingpedia.iot.security-camera/manifest.tt (hunk 7 above)
Comment (greentfrapp) [reply org.thingpedia.iot.security-camera/manifest.tt]: Dropped org.thingpedia.iot.security-camera for now

File: com.nest/camera.js (hunk 3 above)
Comment (gcampax) [reply com.nest/camera.js]: https://developers.nest.com/reference/api-camera

File: org.thingpedia.iot.connectivity/manifest.tt (hunk 4 above)
Comment (gcampax) [reply org.thingpedia.iot.connectivity/manifest.tt]: Yeah let's keep it on hold until we know what it is.

File: io.home-assistant/thermostat.js (hunk 2 above)
Comment (gcampax) [reply io.home-assistant/thermostat.js]: For queries no, but nest also has a set_hvac_state that probably should throw for unsupported values.

File: org.thingpedia.iot.tv/dataset.tt (hunk 14 above)
Comment (gcampax) [reply org.thingpedia.iot.tv/dataset.tt]: No, I don't think so. "on the tv" means "on top of the tv" which is not a command. Nobody will say that.

File: io.home-assistant/thermostat.js (hunk 2 above)
Comment (greentfrapp) [reply io.home-assistant/thermostat.js]: Threw unsupported values in latest commit

File: org.thingpedia.iot.connectivity/manifest.tt (hunk 4 above)
Comment (greentfrapp) [reply org.thingpedia.iot.connectivity/manifest.tt]: Dropped in latest commit

File: org.thingpedia.iot.speaker/manifest.tt (hunk 8 above)
Comment (greentfrapp) [reply org.thingpedia.iot.speaker/manifest.tt]: Changed state() to power(). HA device can be queried about whether it is on/off. This throws an error for Bluetooth.

File: org.thingpedia.iot.tv/dataset.tt (hunk 14 above)
Comment (greentfrapp) [reply org.thingpedia.iot.tv/dataset.tt]: Removed all "on/off my xxx" utterances in tv, speaker, vacuum, switch devices.

File: org.thingpedia.iot.tv/manifest.tt (hunk 15 above)
Comment (greentfrapp) [reply org.thingpedia.iot.tv/manifest.tt]: Changed in devices other than most sensors and vacuum.

File: com.nest/camera.js (hunk 3 above)
Comment (greentfrapp) [reply com.nest/camera.js]: I'm reinstating the old Nest security_camera dataset and manifest for now, since HA doesn't have a camera device.

For now, the Nest security_camera doesn't extend the generic motion sensor yet. Because the Nest security_camera has a `stream  := monitor ((@com.nest.security_camera.current_event()), has_motion == true)` command that sort of overlaps with motion sensor commands.

Comment (gcampax) [reply com.nest/camera.js]: The overlap is the whole point: we need to reannotate potentially conflicting sentences, and we need to decide how to do it.

**Comment by gcampax** [comment]
Ok let's merge this
